### PR TITLE
Add unit tests for pseudo_kl_divergence_loss

### LINKED ISSUE #56: Missing unit tests for `pseudo_kl_divergence` loss function
It appears in `test_dvae_winci020.py` but is not tested in isolation with determinism; related: https://github.com/dwavesystems/dwave-pytorch-plugin/issues/54

### PR DESCRIPTION
**Pull request by abdela47**
This PR adds deterministic unit tests for pseudo_kl_divergence_loss.

The tests cover both documented spin shapes and verify gradient behavior.
They isolate the statistical structure of the loss using deterministic dummy Boltzmann
machines and do not rely on samplers or quantum hardware.

Closes #56.

### CODE REVIEW

File: tests/test_pseudo_kl_divergence.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    out = pseudo_kl_divergence_loss(spins=spins, logits=logits, samples=samples, boltzmann_machine=bm)
+
+    probs = torch.sigmoid(logits)
+    entropy = F.binary_cross_entropy_with_logits(logits, probs)
+    cross_binary = bm.quasi_objective(spins, samples)
+    ref = cross_binary - entropy
+
+    torch.testing.assert_close(out, ref)
+
+def test_pseudo_kl_works_with_3d_spins():
```
[the remark this answers is not in the document]
Comment (abdela47) [reply tests/test_pseudo_kl_divergence.py]: Renamed the test to explicitly reflect the supported 3D spin input shape.

File: tests/test_pseudo_kl_divergence.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+    as documented in the function docstring.
+    """
+    bm = DummyBoltzmannMachine()
+
+    batch, n_samples, n_spins = 3, 5, 4
+    logits = torch.zeros(batch, n_spins)
+
+    # spins: (batch_size, n_samples, n_spins)
+    spins = torch.ones(batch, n_samples, n_spins)
```
[the remark this answers is not in the document]
Comment (abdela47) [reply tests/test_pseudo_kl_divergence.py]: Zero logits are used in the 3D shape test to keep the entropy term simple and stable (p = 0.5), allowing the test to focus purely on documented shape support; nonzero values are covered in the 2D numerical correctness test.

File: tests/test_pseudo_kl_divergence.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+"""
+
+import torch
+import torch.nn.functional as F
+
+from dwave.plugins.torch.models.losses.kl_divergence import pseudo_kl_divergence_loss
+
+class DummyBoltzmannMachine:
+    """
+    Minimal deterministic stand-in for GraphRestrictedBoltzmannMachine.
```
[the remark this answers is not in the document]
Comment (abdela47) [reply tests/test_pseudo_kl_divergence.py]: Refined docstrings to clearly state the tested properties while keeping detailed context at the module level.

**Review by kevinchern — CHANGES_REQUESTED** [review]
Hi @abdela47, just a couple more changes:
1. Can you note [this motivation](https://github.com/dwavesystems/dwave-pytorch-plugin/pull/59#discussion_r2643180962) as a comment in the test?
2. Encapsulate tests in a `unittest` framework and rename tests to be more concise

File: tests/test_pseudo_kl_divergence.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+         [-1,-1, 1, 1,-1, 1],
+         [ 1, 1,-1,-1, 1,-1]],
+        dtype=torch.float32
+        )
+    
+    batch_size, n_spins = spins_data.shape
+    logits = torch.linspace(-2.0, 2.0, steps=batch_size * n_spins).reshape(batch_size, n_spins)
+
+    
+    spins_model = torch.ones(batch_size, n_spins, dtype=torch.float32)
```
Comment (thisac) [inline tests/test_pseudo_kl_divergence.py]:
```suggestion
    
    spins_model = torch.ones(batch_size, n_spins, dtype=torch.float32)
```

File: tests/test_pseudo_kl_divergence.py [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+    out.backward()
+
+    # reference gradient from -entropy only
+    logits2 = logits.detach().clone().requires_grad_(True) 
+    probs2 = torch.sigmoid(logits2)
+    entropy2 = F.binary_cross_entropy_with_logits(logits2, probs2)
+    (-entropy2).backward()
+
+    torch.testing.assert_close(logits.grad, logits2.grad)
```
Comment (thisac) [inline tests/test_pseudo_kl_divergence.py]: Don't forget to add an 
```python
if __name__ == "__main__":
    unittest.main()
```
at the end of this file to make sure the tests are run correctly.

File: tests/test_pseudo_kl_divergence.py [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+        # spins_data: (batch_size, n_spins)
+        spins_data = torch.tensor(
+            [[-1, 1,-1, 1,-1, 1],
+            [ 1,-1, 1,-1, 1,-1],
+            [-1,-1, 1, 1,-1, 1],
+            [ 1, 1,-1,-1, 1,-1]],
+            dtype=torch.float32
+            )
+
```
Comment (kevinchern) [inline tests/test_pseudo_kl_divergence.py]:
```suggestion
```suggestion
        spins_data = torch.tensor(
            [[-1, 1, -1, 1, -1, 1],
             [1, -1, 1, -1, 1, -1],
             [-1, -1, 1, 1, -1, 1],
             [1, 1, -1, -1, 1, -1]],
            dtype=torch.float32
        )

```
```

File: tests/test_pseudo_kl_divergence.py [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+
+    def test_supports_3d_spins(self):
+        """Support 3D spins of shape (batch_size, n_samples, n_spins) as documented."""
+        bm = DummyBoltzmannMachine()
+
+        batch_size, n_samples, n_spins = 3, 5, 4
+        logits = torch.zeros(batch_size, n_spins)
+        # Zero logits are used in the 3D shape test to keep the entropy term simple and stable (p = 0.5), 
+        # allowing the test to focus purely on documented shape support; nonzero values are covered in the
```
Comment (kevinchern) [inline tests/test_pseudo_kl_divergence.py]:
```suggestion
        # Zero logits are used in the 3D shape test to keep the entropy term simple and stable (p = 0.5),
        # allowing the test to focus purely on documented shape support; nonzero values are covered in the
```

File: tests/test_pseudo_kl_divergence.py [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+        This test ensures that pseudo_kl_divergence_loss applies the correct
+        statistical pressure on encoder logits.
+        """
+
+        class ConstantObjectiveBM:
+            def quasi_objective(self, spins_data: torch.Tensor, 
+                                spins_model: torch.Tensor) -> torch.Tensor:
+                # Constant => contributes no gradient wrt logits
+                return torch.tensor(1.2345, dtype=spins_data.dtype, device=spins_data.device)
+
```
Comment (kevinchern) [inline tests/test_pseudo_kl_divergence.py]:
```suggestion
            def quasi_objective(self, spins_data: torch.Tensor,
                                spins_model: torch.Tensor) -> torch.Tensor:
                # Constant => contributes no gradient wrt logits
                return torch.tensor(1.2345, dtype=spins_data.dtype, device=spins_data.device)

```
